IguideME.Web/Controllers/DataController.cs

```diff
@@ -18,23 +23,303 @@ public class DataController : ControllerBase
+            IDictionary<string, int> peers = new Dictionary<string, int>();
+
+            foreach (var entry in data)
+            {
+                data[entry.Key].ToList().ForEach(x =>
+                {
+                    if (!peers.ContainsKey((string)x["studentnaam"]))
+                        peers.Add((string)x["studentnaam"], ((string)x["aanwezig"]).ToLower() == "ja" ? 1 : 0);
+                    else
+                    {
+                        if (((string)x["aanwezig"]).ToLower() == "ja")
+                        {
+                            peers[(string)x["studentnaam"]] += 1;
+                        }
+                    }
+                });
+
+                // add all attendance registries for the student
+                var attendance = data[entry.Key].FirstOrDefault(x => (string) x["studentnaam"] == GetUserName());
+
+                if (attendance != null) registry.Add(attendance);
+            }
+
+            return MakeResponse(registry, peers.Values.ToList());
+        }
+
+        [Authorize]
+        [HttpGet]
+        [Route("/Practice-sessions")]
+        public string PracticeSessionsView()
+        {
+            var data = ReadFile("store/practice_sessions.json");
+            List<object> registry = new List<object>();
+            List<float> grades = new List<float>();
+
+            foreach (var entry in data)
+            {
+                grades.AddRange(data[entry.Key].Select(x => float.Parse((string) x["grade"])));
+
+                // get the practice session for the logged in user
+                var attendance = data[entry.Key].First(x => (string) x["studentnaam"] == GetUserName());
+                registry.Add(attendance);
+            }
+
+            return MakeResponse(registry, grades);
+        }
+
+        [Authorize]
+        [HttpGet]
+        [Route("/Perusall")]
+        public string PerusallView()
+        {
+            var data = ReadFile("store/perusall.json");
+            List<object> assignments = new List<object>();
+            List<float> grades = new List<float>();
+
+            foreach (var entry in data)
+            {
+                grades.AddRange(data[entry.Key].Select(a => float.Parse((string)a["grade"], CultureInfo.InvariantCulture)).ToList());
+
+                // get the perusall assignment of the logged in user
+                var assignment = data[entry.Key].First(x => (string) x["studentnaam"] == GetUserName());
+                assignments.Add(assignment);
+            }
+
+            return MakeResponse(assignments, grades.ToArray());
+        }
+
+        [Authorize]
+        [HttpGet]
+        [Route("/Is-admin")]
+        public Boolean IsAdminView()
+        {
+            return administrators.Contains(GetUser());
+        }
+
+        [Authorize]
+        [HttpPost, HttpGet]
+        [Route("/Admin-perusall")]
+        public string AdminPerusallView()
+        {
+            if (!administrators.Contains(GetUser()))
```

Comment:
At first I had this idea too, but after having consulting with Erwin he told me he'd like to invite teachers to the course so that they can take a look at IguideME. Therefore (for now) there should be a subset of teachers whom have access to the administration panel. I suppose we can replace this eventually with a return statement indicating if the user has the teacher role